Ginger/GingerCoreNET/Repository/SharedRepositoryOperations.cs

```diff
@@ -540,15 +561,33 @@ public static async Task SaveLinkedActivity(Activity LinkedActivity, string Excl
             {
                 var sharedActFullPath = sharedActivity.ContainingFolderFullPath;
                 WorkSpace.Instance.SolutionRepository.MoveSharedRepositoryItemToPrevVersion(sharedActivity);
-                Activity newSharedActivity = (Activity)LinkedActivity.CreateInstance(true);
-                CopyActivityVariableIds(source: sharedActivity, target: newSharedActivity);
-                sharedActivity = newSharedActivity;
-                sharedActivity.Guid = LinkedActivity.ParentGuid;
-                sharedActivity.Type = eSharedItemType.Regular;
-                WorkSpace.Instance.SolutionRepository.AddRepositoryItem(sharedActivity);
-                WorkSpace.Instance.SolutionRepository.MoveItem(sharedActivity, sharedActFullPath);
+                Activity newSharedActivity = (Activity)LinkedActivity.CreateCopy(setNewGUID: false);
+
+                newSharedActivity.Guid = sharedActivity.Guid;
+                newSharedActivity.Type = eSharedItemType.Regular;
+                foreach (Act newActivityAct in newSharedActivity.Acts)
+                {
+                    Act oldActivityAct = (Act)sharedActivity.Acts.FirstOrDefault(a => a.Guid == newActivityAct.ParentGuid);
+                    if (oldActivityAct != null)
+                    {
+                        newActivityAct.Guid = oldActivityAct.Guid;
+                        newActivityAct.ParentGuid = oldActivityAct.ParentGuid;
```

Comment:
Parent GUID should be blank in case of activity is being pushed to shared repository

Reply:
changed the implementation and now the ParentGuid is empty